stakingModel/contracts/InterestDistribution.sol

```diff
@@ -37,28 +37,34 @@ library InterestDistribution {
       * @param _staker                 Staker's address
       * @param _stake                  Amount of stake
       * @param _donationPer            Percentage will to donate.
-      * @param _dailyIntrest           Newly accrued interest.
-      * @param _grandTotalStaked       Total staked by all stakers.
+      * @param _interest               Newly accrued interest since last update.
       *
     */
     function stakeCalculation(
       InterestData storage _interestData, 
       address _staker,
       uint256 _stake, 
       uint256 _donationPer,
-      uint256 _dailyIntrest,
-      uint256 _grandTotalStaked
+      uint256 _interest
       ) 
     internal 
     {
-      _interestData.grandTotalStaked = _interestData.grandTotalStaked.add(_stake);
-      uint currentStake = _interestData.stakers[_staker].stakedToken;
       Staker storage _stakerData = _interestData.stakers[_staker];
+      // Should not update avgYieldRatePerToken for 1st stake as his avg rate should be 0. 
+      if (_interestData.globalTotalStaked > 0) {
+        // Calculating _globalYieldPerToken before updating globalTotalStaked
+        // because the staker still has no part in the interest generated today.
+        // Calculating globalYieldPerToken before the actual update of 
+        // it in the next daily interest accumulation.
+        uint _globalYieldPerToken = getGlobalYieldPerToken(_interest, _interestData.globalTotalStaked);
```

Comment:
getBlobalYieldPerToken is incorrect, see my comment there

Reply:
Replaced with 
updateGlobalYieldPerToken(_interestData, _iTokenRate, _iTokenHoldings);
updateAvgYieldRatePerToken(_stakerData, _interestData.globalYieldPerToken, _stake, _donationPer);